velox/vector/FlatVector.h

```diff
@@ -370,6 +370,35 @@ class FlatVector final : public SimpleVector<T> {
     for (const auto& bufferPtr : stringBuffers_) {
       stringBufferSet_.insert(bufferPtr.get());
     }
+
+    if constexpr (std::is_same_v<T, StringView>) {
+      for (auto i = 0; i < BaseVector::length_; ++i) {
+        if (!BaseVector::isNullAt(i)) {
+          computeStringOffset(rawValues_[i], stringBuffers_);
+        }
+      }
+    }
+  }
+
+  int32_t computeStringOffset(
+      StringView value,
+      const std::vector<BufferPtr>& stringBuffers) {
+    if (value.isInline()) {
+      return 0;
+    }
+
+    int32_t offset = 0;
+    for (const auto& buffer : stringBuffers) {
+      auto start = buffer->as<char>();
+
+      if (value.data() >= start && value.data() < start + buffer->size()) {
```

Comment:
Using `buffer->capacity()` here can avoid the exception.

I can call `setSize` on buffer for this case, but in long term, having both `size` and `capacity` on buffer level is a source of confusion and we probably should keep only one.  The actual size should be tracked uniquely at vector level.  What do you think?

Reply:
@Yuhta Buffer has 2 properties: size and capacity. This is similar to std::vector. Capacity is the number of bytes of memory allocated for the buffer. Size is the logical size of the buffer, i.e. number of bytes that have data. The invariant is size <= capacity.

Reply:
> I can also call setSize on buffer for this case,

That would be great.

Reply:
But in reality the used size is usually tracked at higher level and `Buffer` is just used as a chunk of memory.  In this case having used size tracked at this level is not very convenient.

Reply:
> But in reality the used size is usually tracked at higher level

I'm not sure that's the case. Would you elaborate?

Reply:
BaseVector::size is not related with the sizes of the individual string buffers. BaseVector::size can be used to derive sizes of the nulls and values buffers though.

Reply:
In case of string buffers the invariance is `max(values[i].data() + values[i].size() - buffer.data()) <= Buffer::capacity()`.  Adding `Buffer::size()` is providing very little value here.

We don't need to make the change immediately, it's just something to think about in the long term.

Reply:
Would it be possible to fix TableScan to make sure string buffer sizes are set properly?

Reply:
Yes it's here #4619